### PR TITLE
[6.2.z] Fix Smart Class Parameters - use custom puppet repo

### PR DESCRIPTION
**Pull request by svtkachenko**
Use custom puppet modules repo for puppet related tests instead of puppet forge.
Should fix test failures for CLI/Smart Class Parameters.

One error is because of bug that was fixed in 6.3 but not in 6.2 and `skip_if_bug_open` does not skip that test. What should I do with this test?
```python
% nosetests tests/foreman/cli/test_classparameters.py
.SSSS..........SSSSSS..........E...........
----------------------------------------------------------------------
Ran 43 tests in 901.328s
FAILED (SKIP=10, errors=1)
```

### CODE REVIEW

File: tests/foreman/cli/test_classparameters.py [hunk 1]
```diff
@@ -39,6 +46,8 @@
 )
 from robottelo.test import CLITestCase
 
+from nailgun import entities
```
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: please sort alphabetically

File: tests/foreman/cli/test_classparameters.py [hunk 2]
```diff
@@ -123,27 +132,50 @@ def setUpClass(cls):
...
-        be able to work with unique entity for each specific test. Raise an
-        exception and skip all tests in case not enough parameters returned.
+        be able to work with unique entity for each specific test.
         """
         super(SmartClassParametersTestCase, cls).setUpClass()
-        cls.host_name = settings.server.hostname
-        ssh.command('puppet module install --force puppetlabs/ntp')
-        cls.env = Environment.info({u'name': 'production'})
-        Proxy.importclasses({
+        cls.puppet_module = "robottelo/cli_test_classparameters"
```
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: You never use this variable directly, each call splits the value and uses only part of it. Maybe it's better to define separate variables?

File: tests/foreman/cli/test_classparameters.py [hunk 3]
```diff
@@ -123,27 +132,50 @@ def setUpClass(cls):
...
         cls.sc_params_ids_list = [
-            sc_param['id'] for sc_param in sc_params_list]
+            sc_param['id'] for sc_param in cls.sc_params_list]
+
+    def setUp(self):
+        """Checks that there is at least one not overridden
+        smart class parameter before executing test.
+        """
+        if len(self.sc_params_list) == 0:
+            raise Exception("Not enough smart class parameters. Please"
```
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: nitpick: missing space after word `Please`. Your string will result in `Pleaseupdate puppet module.`

File: tests/foreman/cli/test_classparameters.py [hunk 4]
```diff
@@ -166,11 +198,14 @@ def test_positive_list_parameters_by_host_name(self):
             'id': sc_param_id,
         })
         self.assertEqual(sc_param['override'], True)
+        organization = entities.Organization(id=self.org['id']).search()[0]
```
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: Why do you use searching here? I bet you meant 
```python
organization = entities.Organization(id=self.org['id']).read()
```
But that shouldn't be the issue - you don't even need to read an organization, according to tests the only usage of organization is during Host initialization. nailgun works with entity fields in 2 ways - you can either set them to entity or with entity id. As you already have entity id and there's no other usages of entity - id should be enough:
```python
host = entities.Host(organization=self.org['id']).create()
```

File: tests/foreman/cli/test_classparameters.py (hunk 2 above)
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: Or as an option - a dictionary. With keys `name` and `author`.

File: tests/foreman/cli/test_classparameters.py (hunk 4 above)
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: Comment is applicable to other tests too

File: tests/foreman/cli/test_classparameters.py [hunk 5]
```diff
@@ -123,27 +132,50 @@ def setUpClass(cls):
...
-                               ' work with in provided puppet class')
         cls.sc_params_ids_list = [
-            sc_param['id'] for sc_param in sc_params_list]
+            sc_param['id'] for sc_param in cls.sc_params_list]
+
+    def setUp(self):
+        """Checks that there is at least one not overridden
+        smart class parameter before executing test.
+        """
+        if len(self.sc_params_list) == 0:
```
Comment (abalakh) [inline tests/foreman/cli/test_classparameters.py]: call `super`

**Comment by rplevka** [comment]
@svtkachenko, first of all, thanks for looking into the failures! 

>One error is because of bug that was fixed in 6.3 but not in 6.2 and skip_if_bug_open does not skip that test. What should I do with this test?

Please add it to my list (if it's not yet there): https://github.com/SatelliteQE/robottelo/issues/3868#issuecomment-252241893

Or, if you're brave enough, clone the Bugzilla, for each version (6.1.z, 6.2.z) and update the ID of the bug in the code.

**Comment by abalakh** [comment]
ACK

**Comment by sghai** [comment]
ACK

**Pull request merged**